File: mobile/app/src/main/java/com/data/remote/dto/TaskDto.kt
package com.data.remote.dto

import com.google.gson.annotations.SerializedName

data class TaskDto(
    @SerializedName("id")
    val id: String,
    @SerializedName("userId")
    val userId: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("description")
    val description: String,
    @SerializedName("dueDate")
    val dueDate: String,
    @SerializedName("createdDate")
    val createdDate: String,
    @SerializedName("isCompleted")
    val isCompleted: Boolean
)

File: api/Taskolith.API/Data/Dtos/TaskDto.cs
using Taskolith.API.Data.Types;

namespace Taskolith.API.Data.Dtos;

public record TaskDtoCore(
    Guid TaskId,
    string Title,
    string Description,
    DateTime DueDate,
    DateTime CreatedDate,
    Priority Priority
);

public record TaskDto(
    Guid TaskId,
    string Title,
    string Description,
    int Order,
    DateTime DueDate,
    DateTime CreatedDate,
    bool Completed,
    ICollection<MembershipDto> AssignedMembers,
    Priority Priority
);

public record TaskDtoWithOrganisation(
    TaskDtoCore Task,
    Guid OrganisationId,
    string OrganisationName
);